src/immutable_data.rs

```diff
@@ -35,11 +34,10 @@ pub struct UnpubImmutableData {
 impl UnpubImmutableData {
     /// Creates a new instance of `UnpubImmutableData`
     pub fn new(value: Vec<u8>, owners: PublicKey) -> Self {
```

Comment:
While we are at it, could we please change `owners` to just `owner`? Since we are agnostic to the key type now.
```suggestion
    pub fn new(value: Vec<u8>, owner: PublicKey) -> Self {
```
Same for the field name in struct and the getter function.